### DISCUSSION TITLE
Dynamic arguments in contract write flow

### DISCUSSION THREAD

**Discussion by arlgnv** [wevm/wagmi#857] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/46949445/185053562-20b592c4-7e7e-4c7c-93e1-e84bc502b438.png)
I have this kind of form where on Mint button click I want to invoke write method on contract where I pass two arguments: proof (calculated internally), `tokensNumber` (Amoint field) and tokensCost as value (Total field). How am I supposed to deal with it all? Do I need to each time invoke usePrepareContractWrite when args are changed?

**Comment by jxom** [comment, marked as answer]
I provided an example of how you can use dynamic arguments with `usePrepareContractWrite` here: https://github.com/wagmi-dev/wagmi/discussions/782#discussioncomment-3289133

You can just use `useState` and pass it through to your hook!

**Reply by arlgnv** [reply]
Oh, thanks, it helped. But I have another form in my project (Public sale mint) where on Mint button click, before invoking mint method on contract, I must sign some message and after that invoke write method where I need to pass signedMessage, Here is some example. I found only this one way to do it. Does it seem right?
![image](https://user-images.githubusercontent.com/46949445/185099958-9759e7ef-8b94-4c65-a033-d0f8ff4fe402.png)

**Reply by arlgnv** [reply]
I think that in this case you do not need benefits of `usePrepareContractWrite` because you you'll need to reprepare config anyway due to changing one of parameters.
